src/pages/permissions/panels/GroupOrIdentityChangesTable.tsx

```diff
@@ -0,0 +1,226 @@
+  identities.forEach(
+    (identity) => (identityNameLookup[identity.id] = identity.name),
+  );
+
+  const rows: JSX.Element[] = [];
+  for (const group of groups) {
+    const identityChangesForGroup = groupIdentitiesChangeSummary[group];
+    const addedIdentities: JSX.Element[] = [];
+    const removedIdentities: JSX.Element[] = [];
+
+    for (const identity of identityChangesForGroup.added) {
+      const identityLoggedIn = identity === loggedInIdentityID;
+      addedIdentities.push(
+        <p
+          className="u-no-padding--top modified-item"
+          key={`${group}-${identity}-added`}
+        >
+          +&nbsp;{identityNameLookup[identity]}
+          <Tag isVisible={identityLoggedIn}>You</Tag>
+        </p>,
+      );
+    }
+
+    for (const identity of identityChangesForGroup.removed) {
+      const identityLoggedIn = identity === loggedInIdentityID;
+      removedIdentities.push(
+        <p
+          className="u-no-padding--top modified-item"
+          key={`${group}-${identity}-removed`}
+        >
+          -&nbsp;<span className="removed">{identityNameLookup[identity]}</span>
+          <Tag isVisible={identityLoggedIn}>You</Tag>
+        </p>,
+      );
+    }
+
+    rows.push(
+      <tr key={group} className="modified-row">
+        <td>
+          <p className="u-no-padding--top modified-item">{group}</p>
+        </td>
+        <td>{addedIdentities.concat(removedIdentities)}</td>
+      </tr>,
+    );
+  }
+
+  return rows;
+};
+
+const GroupsOrIdentityChangesTable: FC<Props> = ({
+  groupIdentitiesChangeSummary,
+  identityGroupsChangeSummary,
+  identities,
+  initialGroupBy,
+}) => {
+  const { data: settings } = useSettings();
+  const containerRef = useRef<HTMLDivElement>(null);
+  const [groupBy, setGroupBy] = useState(initialGroupBy);
+  const loggedInIdentityID = settings?.auth_user_name ?? "";
+  const loggedInIdentityModified =
+    loggedInIdentityID in identityGroupsChangeSummary;
+
+  const updateModalTableHeight = () => {
+    const tableContainer = containerRef.current;
+    if (!tableContainer) {
+      return;
+    }
+    // first let the table grow vertically as much as needed
+    tableContainer.setAttribute("style", "height: auto;");
+    const modalMaxHeight = window.innerHeight - 64;
+    const headerHeight = getAbsoluteHeightBelowBySelector(".p-modal__header");
+    const notificationHeight = getAbsoluteHeightBelowById(
+      "current-user-warning",
+    );
+    const footerHeight = getAbsoluteHeightBelowBySelector(".p-modal__footer");
+    const tableHeight = getElementAbsoluteHeight(tableContainer);
+
+    const remainingTableHeight =
+      modalMaxHeight - notificationHeight - headerHeight - footerHeight;
+    if (tableHeight >= remainingTableHeight) {
+      const style = `height: ${remainingTableHeight}px;`;
+      tableContainer.setAttribute("style", style);
+      return;
+    }
+
+    tableContainer.setAttribute("style", "height: auto;");
```

Comment:
Why do we do this twice? Can this be done with scss?

Reply:
The first instance of this will allow the table to grow as large as it needs, so we can get the "real" table height to calculate if it fits. I tried with css but couldn't get it working, especially for reduced viewport height since the table + notification + footer buttons will overflow the modal container causing a double scrollbar.

Reply:
But then we can remove the 2nd one in line 165, no?

Reply:
yeah good point :+1: